.github/actions/configure-tidy/action.yml

```diff
@@ -0,0 +1,37 @@
+name: ./configure
```

Comment:
is special configure for clang-tidy needed? if not, and as clang-tidy is not the fastest, I would say regular apt/configure can be (re)used, for full analysis, all dev packages are needed probably anyway

Reply:
Not for clang-tidy, but for Ubuntu 22.04 - see my first comment at the top of the PR. My 22.04 versions of apt/configure disable lots of features (to sidestep the apt breakage) that could be re-enabled as soon as somebody figures that all out for 22.04 for the "regular" CI builds. What I found most important for now is that the main/Zend directories get checked by clang-tidy. When those code bases are "clean", we can go on to check all those extensions as well.

Reply:
You might need to still filter out few readibility warnings especially case like

```
/home/runner/work/php-src/php-src/ext/date/lib/parse_date.c:12927:20: warning: statement should be inside braces [readability-braces-around-statements]
                        if (yych == 'y') goto yy810;
```

Reply:
But this warning is correct - this piece of code is wrong according to https://github.com/php/php-src/blob/master/CODING_STANDARDS.md

Reply:
I m not saying it s wrong 🙂

1/ this particular case belongs to timelib I believe.
2/ Such warnings make it harder to see more "interesting cases".

But it is just a suggestion 😉

Reply:
The code highlighted is mostly generated code, so I don't think it makes sense to tidy that

Reply:
Yes, tool-generated code is out of your control; but it surprises me that generated code is committed to the git repository?
Most were actually removed 5 years ago by commit f0251a8b16631edf06830e08698cef5d76c1e01b but this one not.

Reply:
@MaxKellermann some of these files were still Git-tracked because of this discussion https://externals.io/message/102796

I don't know if this has changed in the meantime.

The timelib code is maintained elsewhere and in php-src repo it is just bundled. It is bumped with its own release cycle on multiple PHP branches. It's written also in the contributing scheme https://github.com/php/php-src/blob/master/CONTRIBUTING.md#php-source-code-directory-structure

Reply:
we might need, in some places, to filter out via `NOLINT` some macros (e.g. some Zend argument parse ones).